### PR TITLE
Implement sentence concatention

### LINKED ISSUE #8: Create a script to perform Sentence Concatenation
Sentence concatenation, as explored in this paper ([Measuring the Impact of Data Augmentation Methods for Extremely Low-Resource NMT](https://aclanthology.org/2023.loresmt-1.8.pdf)), involves the random concatenation of multiple sentences with <SEP> as a separator token between concatenated sentences. Here, we need a class or script that performs the above for a given language pair. More information can be found in this paper [Sentence Concatenation Approach to Data Augmentation for Neural Machine Translation](https://aclanthology.org/2021.naacl-srw.18.pdf)

> e.g., let x represent english (en) sentences and y represent the yoruba (yor) translated sentences;
> 
> 
> x_1:  how are you my son.
> x_2:  the fuel in the lamp is about to finish.
> 
> 
> y_1: bawo ni omo mi. 
> y_2: epo ti o wa ninu fitila naa ti fẹrẹ pari.
> 
> 
> concatenated sentences
> x: how are you my son. <SEP> the fuel in the lamp is about to finish.
> y: the fuel in the lamp is about to finish. <SEP> epo ti o wa ninu fitila naa ti fẹrẹ pari.

### PR DESCRIPTION
**Pull request by heisguyy**
This commit implement sentence concantention as explained by Kondo et al. There is also a mini implementation of back translation.

Closes #8

### CODE REVIEW

**Comment by heisguyy** [comment]
@Iambusayor @owos @hemhemoh

File: src/utils/concatenate_sentence.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        This parameter is only needed if `do_back_translation` is True. Defaults to None.
+        back_translation_percent (float, optional): percent of target data to back translate.
+        Should take any value between 0 and 1. This parameter is only needed if 
+        `do_back_translation` is True. Defaults to None.
+
+    Returns:
+        Tuple[List[str], List[str]]: Returns of a tuple of the source and target sentences
+    """
+    # load data
+    data = load_dataset("masakhane/mafand", subset_name)
```
Comment (Iambusayor) [inline src/utils/concatenate_sentence.py:83]: There should be an argument that takes the dataset name as an input variable for the script. The default can be `mafand`, but we want the option to set another dataset name. You can use `argparse`.

File: src/utils/concatenate_sentence.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    # separator token. This should ideally be the separator token used by
+    # the model you would train with the output data, not the back translation model.
+    SEPARATOR_TOKEN = "</s>"
+    SENTENCE_LENGTH_THRESHOLD = 10
+    SEQUENTIAL = False
+    DO_BACK_TRANSLATION = True
+    # In a case where you are doing back translation, this parameter should be less
+    # that back_translation_percent * <length of your data>, in this case 33.22(6644 * 0.005)
+    NUMBER_OF_CONCATENATION = 10
+    BACK_TRANSLATION_MODEL = "masakhane/mbart50_yor_en_news"
```
Comment (Iambusayor) [inline src/utils/concatenate_sentence.py:153]: This too should be available for the user to specify

File: src/utils/concatenate_sentence.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    BACK_TRANSLATION_PERCENT = 0.005
```
Comment (Iambusayor) [inline src/utils/concatenate_sentence.py:154]: All these parameters should be available as variables that can be run with the script.

**Review by Iambusayor — CHANGES_REQUESTED** [review]
Great job. Use argparse, so those arguments can be passed as variables when calling the script. Also, add a logger.

File: src/utils/concatenate_sentence.py (hunk 1 above)
Comment (heisguyy) [reply src/utils/concatenate_sentence.py:83]: on it

File: src/utils/concatenate_sentence.py (hunk 2 above)
Comment (heisguyy) [reply src/utils/concatenate_sentence.py:153]: When you say "this", do you mean the number of concatenations or back_translation_model? both of them are customizable already. The code from line 142 down was just an example.

Comment (owos) [inline src/utils/concatenate_sentence.py:153]: Sorry I'm just seeing this, This model and all other masakhane models have been trained on the data we want to use so we can't use this(these) models.

**Review by Iambusayor — APPROVED** [review]

**Pull request merged**